File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    array = []
    result = arr.copy()
    x=0
    while array != result:
        array = result.copy()
        result = []
        for i in array:
            if i%2==0 and i>=50:
                result.append(i//2)
            elif i%2==1 and i<=50:
                result.append(i*2+1)
            else:
                result.append(i)
        x+=1
    
        
    return x-1

File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let arr_prev = [];
    let count = 0;
    while([...arr_prev].toString()!=[...arr].toString()){
        arr_prev = arr.slice();
        for(let i=0;i<arr.length;i++){
            let num = arr[i];
            if(num>=50 && num%2===0){
                arr[i] = num/2;
            }else if(num<50 && num%2!=0){
                arr[i] = (num*2)+1;
            }
        }
        count +=1;
    }
    return count-1;
}